file_writer_control/KafkaTopicUrl.py

```diff
@@ -0,0 +1,21 @@
+import re
+
+
+class KafkaTopicUrl:
+    """
+    Class for extracting address, port and topic name from a Kafka topic url.
+    """
+    test_regexp = re.compile(
+        r"^(\s*(kafka://)?((([^/?#:]+)+)(:(\d+){1,5})?)/([a-zA-Z0-9._-]+)\s*)$"
+    )
+
+    def __init__(self, url: str):
+        result = re.match(KafkaTopicUrl.test_regexp, url)
+        if result is None:
+            raise RuntimeError("Unable to match kafka url.")
+        self.port = 9092
+        if result.group(7) is not None:
+            self.port = int(result.group(7))
+        self.host = result.group(4)
+        self.host_port = f"{self.host}:{self.port}"
+        self.topic = result.group(8)
```

Comment:
Are these magic numbers 7, 4 and 8 shall always hold true, or is there a possibility that the matching pattern subgroup may change?

Reply:
Turns out that there are named (and non-numbered) capture groups. The regex has been updated.

Reply:
Do you see any necessity to raise something if a group is None?

Reply:
I think raising an exception is the easiest and most straightforward way of dealing with an error here.

Reply:
I asked because I am not sure the first exception that is raised if result is None is sufficient, that is do we need extra exceptions for groups?

Reply:
Ah well, if `result != None` only the _port_ group can be `None` as well. It is impossible for the other groups to be `None`.

Reply:
Then we need one more exception and a test. Cool!

Reply:
The _port_ group being `None` is acceptable and is handled by the `if`-statement by falling back to the default _port_ value of 9092.